run.py

```diff
@@ -0,0 +1,187 @@
+    )
+
+    if args.gen_full_results and args.assembly_id:
+        logging.warning(
+            "Generating full results with --gen_full_results while also specifying "
+            "--assembly_id and --input_updated_cif is not recommended. Consider "
+            "parsing a full crystallographic entry with --gen_full_results, or an "
+            "assembly structure with --assembly_id and --input_updated_cif specified."
+        )
+
+    logging.info("input cif: {}".format(input_cif_file))
+    logging.info("output xml: {}".format(args.output_xml))
+    logging.info("output json: {}".format(args.output_json))
+
+    if (
+        args.force
+        or not os.path.exists(interfaces_xml_file)
+        or not os.path.exists(assembly_xml_file)
+    ):
+
+        run_pisalite(
+            input_cif=input_cif_file,
+            xml_output_dir=args.output_xml,
+            pisa_binary=args.pisa_binary,
+            pisa_setup_dir=args.pisa_setup_dir,
+            session_name=args.session_name,
+            asis=args.as_is,
+            lig_proc_mode=args.lig_proc_mode,
+            exclude_ligands=args.exclude_ligands,
+            extended_data=args.gen_full_results,
+        )
+
+    os.makedirs(args.output_xml, exist_ok=True)
+
+    if args.assembly_id:
+        ap = AnalysePisa(
+            pdb_id=args.pdb_id,
+            assembly_id=args.assembly_id,
+            input_updated_cif=args.input_updated_cif,
+        )
+
+        interfaces = ap.interfaces_xml_to_json(
+            assembly_xml_file, interfaces_xml_file, interface_json
+        )
+        ap.assembly_xml_to_json(assembly_xml_file, assembly_json, interfaces)
+
+    if args.gen_full_results:
+
+        data_parser = (
+            ConvertAssemblyXMLToJSON(
+                path_xml=assembly_xml_file,
+                path_json=args.output_json + "/assemblies.json",
+            ),
+            ConvertInterfaceXMLToJSONs(
+                path_xml=interfaces_xml_file,
+                path_jsons=args.output_json + "/interfaces",
+            ),
+            ConvertAssemblyListToJSON(
+                path_txt=args.output_xml
+                + f"/{subdir_extended_data}/assemblies_extended.txt",
+                path_json=args.output_json + f"/{subdir_extended_data}",
+            ),
+            ConvertInterfaceListToJSON(
+                path_txt=args.output_xml
+                + f"/{subdir_extended_data}/interfaces_extended.txt",
+                path_json=args.output_json + f"/{subdir_extended_data}",
+            ),
+            ConvertMonomerListToJSON(
+                path_txt=args.output_xml
+                + f"/{subdir_extended_data}/monomers_extended.txt",
+                path_json=args.output_json + f"/{subdir_extended_data}",
+            ),
+        )
+
+        for parser in data_parser:
+            logging.info(f"Parsing and converting: {parser}")
+            parser.parse()
```

Comment:
Why are we introducing this abstraction instead of having `ConvertAssemblyXMLToJSON` and similar do the parsing in one step?